client/asset/btc/rpcclient.go

```diff
@@ -136,9 +144,64 @@ func (wc *rpcClient) connect(ctx context.Context, _ *sync.WaitGroup) error {
 	return nil
 }
 
+// reconfigure attempts to reconfigure the rpcClient for the new settings. Live
+// reconfiguration is only attempted if the new wallet type is walletTypeRPC. If
+// the special:activelyUsed flag is set, reconfigure will fail if we can't
+// validate ownership of the current deposit address.
+func (wc *rpcClient) reconfigure(cfg *asset.WalletConfig, currentAddress string) (restartRequired bool, err error) {
+	// rpcClient only handles walletTypeRPC.
+	if cfg.Type != walletTypeRPC {
+		restartRequired = true
+		return
+	}
+
+	parsedCfg := new(RPCWalletConfig)
+	if err = config.Unmapify(cfg.Settings, parsedCfg); err != nil {
+		return
+	}
+
+	// Check the RPC configuration.
+	newCfg := &parsedCfg.RPCConfig
+	if err = dexbtc.CheckRPCConfig(&newCfg.RPCConfig, wc.cloneParams.WalletInfo.Name, wc.cloneParams.Network, wc.cloneParams.Ports); err != nil {
+		return
+	}
+
+	// If the RPC configuration has changed, try to update the client.
+	oldCfg := wc.rpcConfig
+	if newCfg.RPCBind != oldCfg.RPCBind ||
+		newCfg.RPCUser != oldCfg.RPCUser ||
+		newCfg.RPCPass != oldCfg.RPCPass ||
+		newCfg.WalletName != oldCfg.WalletName {
+
+		cl, err := newRPCConnection(parsedCfg, wc.cloneParams.SingularWallet)
+		if err != nil {
+			return false, fmt.Errorf("error creating RPC client with new credentials: %v", err)
+		}
+
+		// If the wallet is in active use, check the supplied address.
+		if parsedCfg.ActivelyUsed {
+			method := methodGetAddressInfo
+			if wc.legacyValidateAddressRPC {
+				method = methodValidateAddress
+			}
+			ai := new(GetAddressInfoResult)
+			if err := call(wc.ctx, cl, method, anylist{currentAddress}, ai); err != nil {
```

Comment:
I guess you're not using `ownsAddress` to skip making a `btcutil.Address`?  Seems alright, just stands out.

Reply:
It's because `ownsAddress` is a `rpcClient` method that uses the stored `RawRequesterWithContext`. We don't update the `requesterV` until after this validation.

Reply:
Ahhh, I see now.